### PR TITLE
Save files per region

### PR DESCRIPTION
**Pull request by vlukes**

### CODE REVIEW

File: sfepy/examples/acoustics/vibro_acoustic3d_mid.py [hunk 1]
```diff
@@ -62,15 +62,15 @@
 wc2 = w * c2
 
 regions = {
-    'Gamma0': 'all',
+    'Gamma0_1': 'all',
```
Comment (rc) [inline sfepy/examples/acoustics/vibro_acoustic3d_mid.py:65]: Why this renaming? Also the docstrings use Gamma0.

File: sfepy/discrete/problem.py [hunk 2]
```diff
@@ -831,31 +844,24 @@ def save_state(self, filename, state=None, out=None,
...
-            for var in itervars():
-                rname = var.field.region.name
-                if rname in meshes:
-                    mesh = meshes[rname]
-                else:
-                    mesh = Mesh.from_region(var.field.region, self.domain.mesh,
-                                            localize=True,
-                                            is_surface=var.is_surface)
-                    meshes[rname] = mesh
+            varss = [var for var in itervars()]
```
Comment (rc) [inline sfepy/discrete/problem.py:858]: varsss is unused.

File: sfepy/discrete/problem.py [hunk 3]
```diff
@@ -864,9 +870,25 @@ def save_state(self, filename, state=None, out=None,
...
+                    mesh.write(aux, io='auto', out=vout,
+                            float_format=self.float_format, **kwargs)
+
+        elif file_split_by == 'region':
+            rnames = [v.region_name for v in out.values()]
+            meshes = self.get_meshes_from_region(rnames)
+
+            for rname, mesh in meshes.items():
+                rout = {}
+                out_keys = [key for key, val in out.items()
```
Comment (rc) [inline sfepy/discrete/problem.py:885]: out_keys is unused.

**Comment by rc** [comment]
Thanks! BTW. 
```
sfepy/homogenization/recovery.py:        fpv = pb.conf.options.get('file_per_var', False)
sfepy/homogenization/recovery.py:            pb.save_state(filename, out=outs[ii], file_per_var=fpv)
```
but maybe that is removed in #868?

I wonder about `file_split_by`. The original `file_per_var` also was not very telling that it relates to the results saving. Maybe `split_results_by`?

File: sfepy/examples/acoustics/vibro_acoustic3d_mid.py (hunk 1 above)
Comment (vlukes) [reply sfepy/examples/acoustics/vibro_acoustic3d_mid.py:65]: The same region is named `Gamma0_1` in `vibro_acoustic3d.py`.  `vibro_acoustic3d_mid.py` is subproblem of   `vibro_acoustic3d.py` and the results are merged into the one output file.

File: sfepy/discrete/problem.py (hunk 3 above)
Comment (vlukes) [reply sfepy/discrete/problem.py:885]: ... a piece of code is missing there.

File: sfepy/examples/acoustics/vibro_acoustic3d_mid.py (hunk 1 above)
Comment (rc) [reply sfepy/examples/acoustics/vibro_acoustic3d_mid.py:65]: OK, maybe mention it in the file?

File: sfepy/discrete/problem.py (hunk 3 above)
Comment (vlukes) [reply sfepy/discrete/problem.py]: fixed

**Comment by vlukes** [comment]
> Thanks! BTW.
> 
> ```
> sfepy/homogenization/recovery.py:        fpv = pb.conf.options.get('file_per_var', False)
> sfepy/homogenization/recovery.py:            pb.save_state(filename, out=outs[ii], file_per_var=fpv)
> ```
> 
> but maybe that is removed in #868?

Yes, it will be removed. I update this after merging #868.

> I wonder about `file_split_by`. The original `file_per_var` also was not very telling that it relates to the results saving. Maybe `split_results_by`?

OK, `split_results_by` is good.

**Comment by rc** [comment]
Thanks for the update.

I have tried `sfepy-run sfepy/examples/acoustics/vibro_acoustic3d.py` and it saves 4 files into `output/` - `output/acoustic_wg_Gamma0_1.vtk` contains a subset of data already present in `output/acoustic_wg_mid.vtk`. That is deliberate?

**Comment by vlukes** [comment]
> I have tried `sfepy-run sfepy/examples/acoustics/vibro_acoustic3d.py` and it saves 4 files into `output/` - `output/acoustic_wg_Gamma0_1.vtk` contains a subset of data already present in `output/acoustic_wg_mid.vtk`. That is deliberate?

Actually, it's not intended. But moving all the results from  `output/acoustic_wg_mid.vtk` to `output/acoustic_wg_Gamma0_1.vtk` and suppressing `output/acoustic_wg_mid.vtk` requires a more complex post-processing function.

**Comment by rc** [comment]
OK, LGTM!

**Pull request merged**